### PR TITLE
DOC-3599 Reorganize Learner's Guide

### PR DESCRIPTION
**Pull request by catong**
## [DOC-3599](https://openedx.atlassian.net/browse/DOC-3599)
Improvements to Learner's Guide:
- Re-order the topics in the TOC to match the likeliest workflow, to make info easier to find
- Rename top level sections to make info easier to find
- Reorganize content within topics to make some buried info easier to find, and work in the new order of sections. 
** These changes affected 4 guides: edX and Open edX versions of the Learner's Guide, edX and Open edX versions of the Building and Running Guide.
This PR includes additional work to simplify/clarify files that are included/re-used across guides, remove or modify content that was not applicable to the Open edX versions, etc.

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 

- [x] Doc team review: @edx/doc 

FYI: @dhixonedx, @marcotuts 

### Testing

- [x] Ran ./run_tests.sh without warnings or errors
** 3 warnings in 3 guides are expected due to new anchors that do not clear tests until published on RTD.

### HTML Versions

- [ ] [edX Learner's Guide](http://draft-lg-reorg.readthedocs.io/en/latest/)
- [ ] [Open edX Learner's Guide](http://draftolg-reorg.readthedocs.io/en/latest/)
- [ ] [Building and Running an edX Course Guide](http://draft-br-lg-reorg.readthedocs.io/en/latest/)
- [ ] [Building and Running an Open edX Course Guide](http://draft-obr-for-lg-reorg.readthedocs.io/en/latest/) 

### Post-review
- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [x] Squash commits

### CODE REVIEW

**Comment by grantgoodmanedX** [comment]
The re-organization makes a lot of sense to me. I don't see any pieces that fell out of place in the process. 👍

**Comment by catong** [comment]
@grantgoodmanedX FYI I made two further fixes after your review, both to index files. In the Open edX LG there was an extra line space. And in the edX LG somehow I missed that the "content availability" chapter got moved too far back in the order. I've moved it back up to follow the chapter about course pacing.

File: run_tests.sh [hunk 1]
```diff
@@ -72,11 +72,8 @@ for project in "${projects[@]}"; do
         BUILD_ERRORS=$((BUILD_ERRORS + 1))
     fi
 
-    # Get errors and warnings from the error log
-    IFS=$'\n' read -d '' -r -a error_log < $err_log_file
-
-    num_errors=`echo ${error_log[@]} | grep -o 'ERROR:' | wc -l`
-    num_warnings=`echo ${error_log[@]} | grep -o 'WARNING:' | wc -l`
+    num_errors=`egrep -c 'ERROR:|SEVERE:' < $err_log_file`
+    num_warnings=`egrep -c 'WARNING:' < $err_log_file`
```
Comment (nedbat) [inline run_tests.sh]: This looks like a regression of a change I just made to run_tests.sh

**Review by nedbat — DISMISSED** [review]

**Comment by catong** [comment]
Merging https://github.com/edx/edx-documentation/pull/1449

**Pull request merged**